### ISSUE TITLE
No standard browser under macOS Big Sur

### ISSUE THREAD

**Issue by MartinWe13** [johnste/finicky#146]
Hi everyone,

after the update to macOS Big Sur, Finicky is no longer offered as an option for the standard browser (System Preferences > General)...

...is there a solution for this? :(

Thanks in advance!

Regards
Martin

**Comment by alejandroivan** [comment]
+1 for this, Finicky doesn't appear in the browser list

**Comment by johnste** [comment]
Will update to Big Sur today, hopefully it's an easy fix, though I don't know where to start quite yet.

**Comment by johnste** [comment]
Seems like Finicky will keep being the default browser after an upgrade, but it is indeed not visible in the list.

**Comment by johnste** [comment]
Comparing with [Hammerspoon](https://github.com/Hammerspoon/hammerspoon), finicky doesn't call `LSSetDefaultRoleHandlerForContentType` to register urls and file urls, so that might be the issue. See [this file](https://github.com/Hammerspoon/hammerspoon/blob/2671848973828d6bec08681d4bf107ae69af809d/extensions/urlevent/internal.m)

**Comment by johnste** [comment]
Have tried adding a call to `LSSetDefaultRoleHandlerForContentType`, creating a new release under Big Sur, reading code for Electron and Hammerspoon, and asking in Apple Developer forums, but no luck so far sadly. I really have no idea what's wrong at this point.

**Comment by wad3g** [comment]
As a temporary workaround I installed `defaultbrowser` via homebrew:

- https://formulae.brew.sh/formula/defaultbrowser
- https://github.com/kerma/defaultbrowser

Everything appears to be working like normal for now.

**Comment by johnste** [comment]
Making some progress – replacing the plist file with one from Hammerspoon makes Finicky appear as a choice again. Now to figure out what part of it makes it work, will take a while to test. 

![image](https://user-images.githubusercontent.com/886051/99722969-cc4b5880-2ab1-11eb-9f9b-3d1d996a3ba1.png)

**Comment by alejandroivan** [comment]
Seems like adding this:
```
<dict>
    <key>CFBundleTypeExtensions</key>
    <array>
        <string>xhtml</string>
        <string>xht</string>
        <string>xhtm</string>
        <string>xht</string>
    </array>
    <key>CFBundleTypeIconFile</key>
    <string>finicky</string>
    <key>CFBundleTypeMIMETypes</key>
    <array>
        <string>application/xhtml+xml</string>
    </array>
    <key>CFBundleTypeName</key>
    <string>XHTML document</string>
    <key>CFBundleTypeRole</key>
    <string>Viewer</string>
    <key>LSTypeIsPackage</key>
    <integer>0</integer>
    <key>NSDocumentClass</key>
    <string>NSDocument</string>
</dict>
```

under `CFBundleDocumentTypes` fixes the issue.

**Comment by johnste** [comment]
Indeed, seems like that does the trick. Thank you! Will try to get release out today.

**Comment by johnste** [comment]
Fixed in v3.1.3 https://github.com/johnste/finicky/releases/tag/v3.1.3

**Comment by alejandroivan** [comment]
Awesome! Thank you!
It would be nice if @waldirious updates the Caskroom definition so we can upgrade using Caskroom :)
